src/main/java/org/apache/commons/lang3/Validate.java

```diff
@@ -874,7 +888,752 @@ public static void matchesPattern(final CharSequence input, final String pattern
+     * @throws IllegalArgumentException if the value is not a number
+     * @see #notNaN(double) 
+     * 
+     * @since 3.4
+     */
+    public static void notNaN(final double value, final String message, final Object... values) {
+        if (value != value) {
+            throw new IllegalArgumentException(String.format(message, values));
+        }
+    }
+    
+    // finite
+    //---------------------------------------------------------------------------------
+    
+    /**
+     * <p>Validates that the specified argument is not infinite or {@code NaN};
+     * otherwise throwing an exception.</p>
+     * 
+     * <pre>Validate.finite(myDouble);</pre>
+     * 
+     * <p>The message of the exception is &quot;The value is invalid: %f&quot;.</p>
+     * 
+     * @param value  the value to validate
+     * @throws IllegalArgumentException if the value is infinite or {@code NaN}
+     * @see #finite(double, java.lang.String, java.lang.Object...) 
+     * 
+     * @since 3.4
+     */
+    public static void finite(final double value) {
+        finite(value, DEFAULT_FINITE_EX_MESSAGE, value);
+    }
+    
+    /**
+     * <p>Validates that the specified argument is not infinite or {@code NaN};
+     * otherwise throwing an exception with the specified message.</p>
+     * 
+     * <pre>Validate.finite(myDouble, "The argument must contain a numeric value");</pre>
+     *
+     * @param value the value to validate
+     * @param message  the {@link String#format(String, Object...)} exception message if invalid, not null
+     * @param values  the optional values for the formatted exception message
+     * @throws IllegalArgumentException if the value is infinite or {@code NaN}
+     * @see #finite(double) 
+     * 
+     * @since 3.4
+     */
+    public static void finite(final double value, final String message, final Object... values) {
+        if (value != value || Double.isInfinite(value)) {
+            throw new IllegalArgumentException(String.format(message, values));
+        }
+    }
+    
+    // greater
+    //---------------------------------------------------------------------------------
+    
+    /**
+     * <p>Validates that the specified argument is strictly greater than a given
+     * reference; otherwise throwing an exception.</p>
+     * 
+     * <pre>Validate.greaterObj(myObject, refObject);</pre>
+     * 
+     * <p>The message of the exception is &quot;The value {@code value} is not 
+     * greater than {@code min}&quot;.</p>
+     * 
+     * @param <T>  the type of the argument object
+     * @param min  the reference value
+     * @param value  the object to validate
+     * @throws IllegalArgumentException if {@code value} is smaller than or equal to {@code min}
+     * @see #greaterObj(java.lang.Object, java.lang.Comparable, java.lang.String, java.lang.Object...) 
+     * 
+     * @since 3.4
+     */
+    public static <T> void greaterObj(final T min, final Comparable<T> value) {
```

Comment:
(1)- If I call the method 'greater' I'm creating an ambiguous definition as longs and double can be boxed into comparable objects.

(2)- I agree about swapping the arguments, I wanted to do it the other way around from the start, but if you look at the methods inclusiveBetween and exclusiveBetween, they take the reference values first. I thought it'd be better to make my methods homogeneous with those. Can you confirm that you want me to make this change?

I don't have a fix for point (1).